### ISSUE TITLE
"Unknown unit" HTTP API response for step durations in 'd', 'w' or 'y'

### ISSUE THREAD

**Issue by mactyr** [VictoriaMetrics/VictoriaMetrics#641]
**Describe the bug**
When I query `/api/v1/query_range/` from the VictoriaMetrics HTTP API with a `step` query parameter defined in units of `d`, `w`, or `y` I receive a response with status 422 and an error message like `cannot parse "step"="1d": time: unknown unit d in duration 1d`. The equivalent duration defined in hours instead (e.g. `step=24h`) works fine.

**To Reproduce**
I receive this response for any `query_range` query I make with a step duration in units of `d`, `w` or `y`, but not for `h` or `m`.

**Expected behavior**
Units of `d` return data with a step of the specified number of days, etc., per the Prometheus definition of `<duration>` in [their API doc](https://prometheus.io/docs/prometheus/latest/querying/api/).

**Version**
```
$ ./victoria-metrics-prod --version
victoria-metrics-20200714-141946-tags-v1.38.1-0-gb442a42d8
```

**Bonus question**
Assuming that 'days' are intended to be a supported duration, is there any way to query days aligned with a local timezone? It looks like I can use `offset` in the query to align 24 hour periods with the day boundaries for a particular UTC offset, but the part I'm not sure about is dealing with daylight savings time, where there are occasional calendar days that contain 23 or 25 hours, and different UTC offsets on either side of the transition days. (For our application we often need to view data aligned with the local timezone of the device, though it arrives stamped in UTC.)

**Comment by valyala** [comment]
@mactyr , thanks for the bug report! It should be fixed in the commit 4f526cc81677bf3e897571ca7adacd6b05f1d044 . This commit will be included in the next release of VictoriaMetrics. In the mean time you can build VictoriaMetrics from the commit mentioned above according to [build instructions](https://victoriametrics.github.io/#how-to-build-from-sources) and verify whether it works as expected.

>  is there any way to query days aligned with a local timezone?

VictoriaMetrics operates only with timestamps in UTC. All the adjustments to other time zones must be performed on client side.

**Comment by mactyr** [comment]
Thank you, built from source and confirmed that I can now use `d`, `w` and `y` steps.

**Comment by valyala** [comment]
FYI, the bugfix has been included in [v1.39.0 release](https://github.com/VictoriaMetrics/VictoriaMetrics/releases).
